src/Stratis.Features.FederatedPeg.IntegrationTests/NodeInitialisationTests.cs

```diff
@@ -0,0 +1,194 @@
+            }
+        }
+
+        [Fact]
+        public async Task GatewayPairStarts()
+        {
+            using (SidechainNodeBuilder nodeBuilder = SidechainNodeBuilder.CreateSidechainNodeBuilder(this))
+            {
+                Key federationKey = new Key();
+
+                CoreNode side = nodeBuilder.CreateSidechainFederationNode(this.sidechainNetwork, this.mainNetwork, federationKey);
+                side.AppendToConfig("sidechain=1");
+                side.AppendToConfig($"redeemscript={this.scriptAndAddresses.payToMultiSig}");
+                side.AppendToConfig($"publickey={this.sidechainNetwork.FederationMnemonics[0].DeriveExtKey().PrivateKey.PubKey}");
+                side.AppendToConfig("federationips=0.0.0.0,0.0.0.1"); // Placeholders
+                side.AppendToConfig($"mindepositconfirmations={DepositConfirmations}");
+
+                CoreNode main = nodeBuilder.CreateMainChainFederationNode(this.mainNetwork, this.sidechainNetwork).WithWallet();
+                main.AppendToConfig("mainchain=1");
+                main.AppendToConfig($"redeemscript={this.scriptAndAddresses.payToMultiSig}");
+                main.AppendToConfig($"publickey={this.sidechainNetwork.FederationMnemonics[0].DeriveExtKey().PrivateKey.PubKey}");
+                main.AppendToConfig("federationips=0.0.0.0,0.0.0.1"); // Placeholders
+                main.AppendToConfig($"mindepositconfirmations={DepositConfirmations}");
+
+                side.AppendToConfig($"counterchainapiport={main.ApiPort}");
+                main.AppendToConfig($"counterchainapiport={side.ApiPort}");
+
+                side.Start();
+                main.Start();
+
+                Assert.Equal(CoreNodeState.Running, main.State);
+                Assert.Equal(CoreNodeState.Running, side.State);
+
+                $"http://localhost:{side.ApiPort}/api".AppendPathSegment("FederationWallet/enable-federation").PostJsonAsync(new EnableFederationRequest
+                {
+                    Password = "password",
+                    Mnemonic = this.sidechainNetwork.FederationMnemonics[0].ToString()
+                }).Result.StatusCode.Should().Be(HttpStatusCode.OK);
+
+                $"http://localhost:{main.ApiPort}/api".AppendPathSegment("FederationWallet/enable-federation").PostJsonAsync(new EnableFederationRequest
+                {
+                    Password = "password",
+                    Mnemonic = this.sidechainNetwork.FederationMnemonics[0].ToString()
+                }).Result.StatusCode.Should().Be(HttpStatusCode.OK);
+
+                // If one node progresses far enough that the other can advance it's NextMatureDepositHeight, they're talking!
+                TestHelper.MineBlocks(main, DepositConfirmations + 1);
+                TestBase.WaitLoop(() => side.FullNode.NodeService<ICrossChainTransferStore>().NextMatureDepositHeight > 0);
```

Comment:
Should this test also happen for the main node - i.e. test for bi-directional communication?

Reply:
It should. Unfortunately that's a bit harder as `TestPoAMiner` doesn't know how to deal with adding the collateral parts to the block yet I think. Will keep trying.

Reply:
Yeah, I've had a little dive and that will take a couple hours more work, as the nodes are checking for commitments in the blocks even when there's no collateral required. @noescape00 might have some ideas about how to easily deal with this?